core/src/main/java/org/infinispan/reactive/publisher/impl/LocalPublisherManagerImpl.java

```diff
@@ -241,18 +241,18 @@ private SegmentAwarePublisherImpl(IntSet segments, CacheSet<I> set,
 
       @Override
       public void subscribe(Subscriber<? super R> s, IntConsumer completedSegmentConsumer, IntConsumer lostSegmentConsumer) {
-         Flowable<Publisher<R>> segmentPublishers;
+         Flowable<R> resultPublisher;
          switch (deliveryGuarantee) {
             case AT_MOST_ONCE:
-                segmentPublishers = Flowable.fromStream(segments.intStream().mapToObj(segment -> {
-                   Publisher<I> publisher = set.localPublisher(segment);
-                   if (predicate != null) {
-                      publisher = Flowable.fromPublisher(publisher)
-                            .filter(predicate);
-                   }
+               resultPublisher = Flowable.fromStream(segments.intStream().mapToObj(segment -> {
```

Comment:
Do we really need to use streams here?
Wouldn't it be better to use `Flowable.fromIterable(segments).concatMap(segment -> ...)`?
(Ideally on separate lines so the `.` on line 286 can be aligned to the previous `.` as well.)

Reply:
Sure, I can add that change as part of this PR.